File: src/extras-attica/libhomepageentry.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_Attica__HomePageEntry
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QUrl>
#include <homepageentry.h>
#include "libhomepageentry.h"
#include "libhomepageentry.hxx"

Attica__HomePageEntry* Attica__HomePageEntry_new() {
    return new Attica::HomePageEntry();
}

Attica__HomePageEntry* Attica__HomePageEntry_new2(const Attica__HomePageEntry* other) {
    return new Attica::HomePageEntry(*other);
}

void Attica__HomePageEntry_OperatorAssign(Attica__HomePageEntry* self, const Attica__HomePageEntry* other) {
    self->operator=(*other);
}

libqt_string Attica__HomePageEntry_Type(const Attica__HomePageEntry* self) {
    QString _ret = self->type();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void Attica__HomePageEntry_SetType(Attica__HomePageEntry* self, const libqt_string typeVal) {
    QString typeVal_QString = QString::fromUtf8(typeVal.data, typeVal.len);
    self->setType(typeVal_QString);
}

QUrl* Attica__HomePageEntry_Url(const Attica__HomePageEntry* self) {
    return new QUrl(self->url());
}

void Attica__HomePageEntry_SetUrl(Attica__HomePageEntry* self, const QUrl* url) {
    self->setUrl(*url);
}

void Attica__HomePageEntry_Delete(Attica__HomePageEntry* self) {
    delete self;
}

File: src/extras-attica/libhomepageentry.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const std = @import("std");

/// https://api.kde.org/attica-homepageentry.html
pub const attica__homepageentry = struct {
    /// New constructs a new Attica::HomePageEntry object.
    ///
    ///
    pub fn New() QtC.Attica__HomePageEntry {
        return qtc.Attica__HomePageEntry_new();
    }

    /// New2 constructs a new Attica::HomePageEntry object.
    ///
    /// ``` other: QtC.Attica__HomePageEntry ```
    pub fn New2(other: ?*anyopaque) QtC.Attica__HomePageEntry {
        return qtc.Attica__HomePageEntry_new2(@ptrCast(other));
    }

    /// [Qt documentation](https://api.kde.org/attica-homepageentry.html#operator-eq)
    ///
    /// ``` self: QtC.Attica__HomePageEntry, other: QtC.Attica__HomePageEntry ```
    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.Attica__HomePageEntry_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://api.kde.org/attica-homepageentry.html#type)
    ///
    /// ``` self: QtC.Attica__HomePageEntry, allocator: std.mem.Allocator ```
    pub fn Type(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.Attica__HomePageEntry_Type(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("attica::homepageentry.Type: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/attica-homepageentry.html#setType)
    ///
    /// ``` self: QtC.Attica__HomePageEntry, typeVal: []const u8 ```
    pub fn SetType(self: ?*anyopaque, typeVal: []const u8) void {
        const typeVal_str = qtc.libqt_string{
            .len = typeVal.len,
            .data = typeVal.ptr,
        };
        qtc.Attica__HomePageEntry_SetType(@ptrCast(self), typeVal_str);
    }

    /// [Qt documentation](https://api.kde.org/attica-homepageentry.html#url)
    ///
    /// ``` self: QtC.Attica__HomePageEntry ```
    pub fn Url(self: ?*anyopaque) QtC.QUrl {
        return qtc.Attica__HomePageEntry_Url(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/attica-homepageentry.html#setUrl)
    ///
    /// ``` self: QtC.Attica__HomePageEntry, url: QtC.QUrl ```
    pub fn SetUrl(self: ?*anyopaque, url: ?*anyopaque) void {
        qtc.Attica__HomePageEntry_SetUrl(@ptrCast(self), @ptrCast(url));
    }

    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.Attica__HomePageEntry ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.Attica__HomePageEntry_Delete(@ptrCast(self));
    }
};
